### ISSUE TITLE
Method reference to Enum::name reported as rawtype.

### ISSUE THREAD

**Issue by sdavids** [google/error-prone#569]
```
import java.nio.file.StandardOpenOption;
import java.util.Arrays;

import static java.util.stream.Collectors.joining;

public class Test {
    public static void main(String[] args) {
        Arrays.stream(StandardOpenOption.values())
                .map(Enum::name)                      // (1) 
                .collect(joining(","));

        Arrays.stream(StandardOpenOption.values())
                .map(Enum<StandardOpenOption>::name)  // (2) 
                .collect(joining(","));
    }
}
```

Variant 1 is reported by errorprone:

```
Test.java:9: warning: [rawtypes] found raw type: Enum
                .map(Enum::name).collect(joining(","));
                     ^
  missing type arguments for generic class Enum<E>
  where E is a type-variable:
    E extends Enum<E> declared in class Enum
1 warning
```

Variant 2 is reported by IntelliJ inspection:

"Java | Verbose or redundant code constructs | Redundant type arguments".

Bug or feature?

**Comment by cushon** [comment]
This is a javac bug: https://bugs.openjdk.java.net/browse/JDK-8134009, `Enum::name` is correct and should not result in a warning.

**Comment by cushon** [comment]
Oops, that was the wrong bug. This is https://bugs.openjdk.java.net/browse/JDK-8063054.

**Comment by cushon** [comment]
https://bugs.openjdk.java.net/browse/JDK-8063054 was fixed in JDK 10.
